backend/LexBoxApi/Services/HgService.cs

```diff
@@ -75,6 +75,39 @@ private void InitRepoAt(string code)
         );
     }
 
+    public async Task PrepareEmptyRepo(string code, string tempRepoSuffix)
+    {
+        var tempRepoName = $"{code}__{tempRepoSuffix}";
+        await Task.Run(() =>
+        {
+            var deletedRepoPath = Path.Combine(_options.Value.RepoPath, DELETED_REPO_FOLDER);
+            var directory = Directory.CreateDirectory(deletedRepoPath);
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                directory.UnixFileMode = Permissions;
+            var dest = Directory.CreateDirectory(Path.Combine(directory.FullName, tempRepoName));
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                dest.UnixFileMode = Permissions;
+            FileUtils.CopyFilesRecursively(
+                new DirectoryInfo("Services/HgEmptyRepo"),
+                dest,
+                Permissions
+            );
+        });
+    }
+
+    public async Task MoveEmptyRepoIntoPlace(string code, string tempRepoSuffix)
+    {
+        var tempRepoName = $"{code}__{tempRepoSuffix}";
+        await Task.Run(() =>
```

Comment:
This is the step we want to happen as fast as possible. I wonder if that's a reason to avoid the small overhead of `Task.Run()`. Probably not very important.

Reply:
Done in commit b2f6ef07.